### ISSUE TITLE
redundant-clone: false positive with assert_eq!()

### ISSUE THREAD

**Issue by djc** [rust-lang/rust-clippy#5973]
I tried this code:

```rust
fn main() {
    let foo = Some("bar".to_string());
    assert_eq!(foo.clone(), None, "not equal {}", foo.unwrap());
}
```

I expected to see this happen: no lints

Instead, this happened: got a `redundant-clone` warning.

This seems similar to #5700 or #5595, but maybe not quite the same?

### Meta

- `cargo clippy -V`: 0.0.212 (2020-08-26 2d8a3b9)
- `rustc -Vv`:

```
rustc 1.45.2 (d3fb005a3 2020-07-31)
binary: rustc
commit-hash: d3fb005a39e62501b8b0b356166e515ae24e2e54
commit-date: 2020-07-31
host: x86_64-apple-darwin
release: 1.45.2
LLVM version: 10.0
```

(I wonder if the number of false positives reported here are contributing to redudant clone also being relatively high on the commonly ignored lints list from #5418.)

**Comment by flip1995** [comment]
> (I wonder if the number of false positives reported here are contributing to redudant clone also being relatively high on the commonly ignored lints list from #5418.)

That is probably because the `redundant_clone` lint was riddled with FPs before the rewrite to a MIR lint, which got rid of even more annoying FPs than the ones that occur now.
